core/src/main/java/org/testcontainers/dockerclient/RootlessDockerClientProviderStrategy.java

```diff
@@ -0,0 +1,63 @@
+package org.testcontainers.dockerclient;
+
+import com.sun.jna.Library;
+import com.sun.jna.Native;
+import org.apache.commons.lang.SystemUtils;
+
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ *
+ * @deprecated this class is used by the SPI and should not be used directly
+ */
+@Deprecated
+public final class RootlessDockerClientProviderStrategy extends DockerClientProviderStrategy {
+
+    public static final int PRIORITY = EnvironmentAndSystemPropertyClientProviderStrategy.PRIORITY + 20;
```

Comment:
Do we want this to be the highest priority? I'd put this lower than `UnixSocketClientProviderStrategy`, at least after #1771 is merged (so that `DOCKER_HOST` is always respected, non-rootless-docker is tried first, and automatically configured rootless docker is tried last of all)

Reply:
I didn't want rootless DOCKER_HOST to take precedence (because we need to override the host ip) but yeah, I see what you mean.

I will make it have lower priority, but ensure that EnvAndSysPropsStrategy does not detect rootless host that easily

Reply:
update:
with overridable host, it is no longer a problem when `EnvAndSysPropsStrategy` detects the rootless `DOCKER_HOST`